IO/Ex1.cpp

```diff
@@ -0,0 +1,108 @@
+
+#include <stdexcept> //runtime_error
+#include <cassert>
+
+#include "Ex1.hpp"
+
+/************************************* Main Functions *******************************************/
+
+//Ctor
+FileHandler::FileHandler(const std::string path, std::ios_base::openmode permissions)
+: m_path(path), 
+  m_permissions(permissions)
+{
+
+    m_file.open(path, permissions);
+    if (!m_file.is_open()) {
+        throw std::runtime_error("Failed to open file: " + path);
+    }
+}
+
+//CCtor
+FileHandler::FileHandler(FileHandler &other) 
```

Comment:
please add a comment about why this copy consturctor receives non const reference

Reply:
added in the header